### ISSUE TITLE
FR: Allow traffic from ALL apps in Pause mode

### ISSUE THREAD

**Issue by iulko** [celzero/rethink-app#1364]
When I turn off Rethink using Pause in Lockdown mode the apps dont connect to internet

**Comment by ignoramous** [comment]
Apps that are *blocked* (wifi or mobile) will continue to be blocked even in *pause* mode, but if Rethink was also in *Lockdown* (aka `Block connections without VPN` turned ON), then other non-blocked apps continued to be routed through the tunnel.

What are these apps you're unable to connect to?

- *Pause* Rethink (assuming it is already *Lockdown*)
- Open any app that's *not* blocked and see if it is able to access Internet.
- *Unpause* Rethink
- Go to *Network Log* and search for entries of the non-blocked app above.
- Tap on those entries and in the *bottomsheet* that comes up, check if the connection was "blocked" or "allowed" or failed to connect (look at the *footer* of the *bottomsheet* for final connection status).

**Comment by ignoramous** [comment]
> So what pause does in lockdown mode exactly?

*Pause* has always blocked existing *blocked* apps (with or without *Lockdown*).

To allow a *blocked* app to be able to connect, you'd have to remove those rules (regardless of whether the VPN is *Lockdown* or not), as *Pause* won't cut it. 

In *Pause* mode, *allowed* apps use:
- `System DNS` (without any rules applied).
- Underlying network (wifi/mobile) directly (if Rethink is not *Lockdown*) or Rethink's tunnel network (if Rethink is *Lockdown*).
- Universal (global) & per-app IP/domain rules are skipped.

**Comment by ignoramous** [comment]
> So in lockdown it just changed apps from "Allowed" to "Bypass" essentially?

No, *Bypass* doesn't skip *per-app* rules, nor does it use `System DNS`.

> But maybe we could have an option/setting (which on/off would be saved) inside pause menu, like a tickbox "Temporary allow all apps" which would also use RULE8 for lockdown but for all apps

I see this setting as a foot-gun. Most apps I block (calculator, flash light, govt apps), I want them to remain blocked; and unblocking them ALL in *Pause* mode is a foot-gun (just so one other app that's updating something would work). *Pausing* individual apps is unsupported, and so, the closest you come to that UX is, unblocking said app, getting work done, then blocking it back again. Way better than unblocking all blocked apps, even if temporarily.

I'll think about it some more, but as of today, I am not inclined to allowing ALL apps in *Pause* mode.
